cpu/stm32_common/cpu_init.c

```diff
@@ -49,6 +49,20 @@ void cpu_init(void)
     periph_clk_en(APB1, BIT_APB_PWREN);
     /* initialize the system clock as configured in the periph_conf.h */
     stmclk_init_sysclk();
+
+#if defined(CPU_FAM_STM32L1)
```

Comment:
This approach is valid for the whole stm32 family AFAIK

Reply:
It's valid, but not needed at least on L1 and L4 - they have GPIO configured as analog inputs by default (see default MODER value in the Reference Manual).

Not sure about F1 (not an energy-efficient series anyway), but probably newer F-series don't need it either.